### PR TITLE
Basic config support

### PR DESCRIPTION
**Pull request by gplsteph**
- Made the Maya publisher hook usable with a basic config by allowing the publish template to be optional: if not set, output path are derived from the current Maya workspace.
- Fixed a problem preventing the framework to be loaded with Python2 (e.g. in Maya) because we only build PySide2 binaries for Python3 (PySide2 for Python 2 is not pip installable). We now check if Qt is already available before attempting to activate our PySide package, which, generally speaking, seems a sensible thing to do.
- If an FBX was generated and published by the Publish FBX item, it is now re-used by the turntable publisher, instead of exporting a new temp FBX. A dependency is made between the FBX publish and the turntable publish.
- Users settings are now saved and restored.
- Some polish in a couple of places.

### CODE REVIEW

File: hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py [hunk 1]
```diff
@@ -450,7 +461,7 @@ def settings(self):
             },
             "Turntable Assets Path": {
                 "type": "string",
-                "default": "/Game/maya_turntable_assets/",  # TODO: make sure the trailing / is not needed.
+                "default": "/Game/maya_turntable_assets",
```
Comment (000paradox000) [inline hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py:464]: If this hardcoded string is needed can we manage constants variables/attributes in the class?

File: hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py [hunk 2]
```diff
@@ -562,13 +573,60 @@ def set_ui_settings(self, widget, settings):
...
 
+    def load_saved_ui_settings(self, settings):
+        """
+        Load saved settings and update the given settings dictionary with them.
+
+        :param settings: A dictionary where keys are settings names and
+                         values Settings instances.
+        """
+        # Retrieve SG utils framework settings module and instantiate a manager
+        fw = self.load_framework("tk-framework-shotgunutils_v5.x.x")
```
Comment (000paradox000) [inline hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py:597]: Is possible to manage frameworks instances as an attribute of the class that is created once?

**Review by 000paradox000 — COMMENTED** [review]
LGTM just small observations

File: hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py (hunk 2 above)
Comment (gplsteph) [reply hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py:597]: I don't think there is any guarantee you'll get the same instance of the TK Hook. If you want to persist values, you need to store them in properties.

File: hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py (hunk 1 above)
Comment (gplsteph) [reply hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py:464]: Could be, but I think it is only set here and TK Hook default settings are usually defined like that.

**Review by 000paradox000 — APPROVED** [review]
LGTM

File: hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py (hunk 2 above)
Comment (000paradox000) [reply hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py:597]: Thanks

File: hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py (hunk 1 above)
Comment (000paradox000) [reply hooks/tk-multi-publish2/tk-maya/basic/publish_turntable.py:464]: Thanks

**Review by josh-t — APPROVED** [review]

**Pull request merged**